content/blog/accessing-hpe-data-fabric-s3-storage-from-spring-boot-s3-micro-service-deployed-in-k3s-cluster.md

```diff
@@ -0,0 +1,147 @@
+
+**Spring Boot**
+
+Spring Boot is an open source Java-based framework used to create a micro Service. Many Real world applications are written in Spring Boot for faster development and better maintainability. More information can be found at** [spring.io](https://spring.io/projects/spring-boot)**
+
+Note: We can move the Spring Boot application to Quarkus with no change in code.  Moving code to Quarkus will reduce the footprint of Spring Boot application. More information can be found at [Quarkus site](https://quarkus.io/blog/quarkus-for-spring-developers/)
+
+In this blog, checkout existing Spring Boot application from [GitHub](https://github.hpe.com/kiran-mavatoor/df-s3-springboot-k3s-demo) , customise it and execute. 
+
+**Application Prerequisites**
+
+1.      K3S cluster must be accessible. Note down the control plan node details. This information is required to deploy the Spring Boot application.
+
+2.      Access the HPE Data Fabric Object Store service UI running on port 9000. For example URL  - https://FQDN:9000/ .  Note down the access key and secret key. Default access key and secret key is “minioadmin” and “minioadmin” respectively. It is advised to change the default values.
+
+3.      Java 11 , Apache Maven 3.8+ , Docker Client.
+
+**Build and Install Steps**
+
+1.      Check out  an existing Spring Boot application from \[github](<https://github.hpe.com/kiran-mavatoor/df-s3-springboot-k3s-demo>).
+
+2.      Copy the ssl_usertruststore.p12 from the HPE Data Fabric cluster into certs folder under project directory. The ssl_usertruststore.p12 file is located at /opt/mapr/conf directory in cluster node. The password for p12 can be copied from “ssl.client.truststore.password” property value in /opt/mapr/conf/ssl-client.xml .
+
+3.      From project directory, open resources/application.properties. Change the key values as per your environment. 
+
+4.      Execute “mvn clean install” .
+
+5.      The distributable is available in target/df-s3-springboot-k3s-demo-1.0-SNAPSHOT.jar .
+
+6.      Edit the DockerFile located in project directory. The value of 
+
+“-Djavax.net.ssl.trustStorePassword”  must be same as the value of  “ssl.client.truststore.password” obtained from step 2. 
+
+Note: This value can be configured using config-map.yaml in K3S cluster.
+
+7.      Execute below docker commands to build docker image and push it to docker hub. 
+
+Note: Alternatively , we can use podman instead of docker to create images. More information on podman can be obtained from [here](https://developers.redhat.com/blog/2020/11/19/transitioning-from-docker-to-podman#transition_to_the_podman_cli) [](https://developers.redhat.com/blog/2020/11/19/transitioning-from-docker-to-podman#transition_to_the_podman_cli)
+
+```
```

Comment:
For a better rendering of the code snippet, please make sure to specify a language syntax after the triple backticks at the beginning of the code block snippet.

```dockerfile
docker build -f Dockerfile -t <Dockerhub user id>/df-s3-springboot-k3s-demo:latest .
docker image ls
docker login -u <Dockerhub user id>
> enter password:  <Dockerhub password>
docker push <Dockerhub user id>/df-s3-springboot-k3s-demo:latest

```